app/scripts/controllers/transactions/index.js

```diff
@@ -2746,30 +2187,21 @@ export default class TransactionController extends EventEmitter {
     return gasValuesInGwei;
   }
 
-  _failTransaction(txId, error, actionId) {
+  _failTransaction(txId, error) {
     this.txStateManager.setTxStatusFailed(txId, error);
     const txMeta = this.txStateManager.getTransaction(txId);
-    this._trackTransactionMetricsEvent(
-      txMeta,
-      TransactionMetaMetricsEvent.finalized,
-      actionId,
```

Comment:
Just for audit sake, can I confirm in these instances where we've removed the `actionId` argument, are we certain it's already been persisted in the transaction metadata and so can still be accessed by the client?

If so, it raises the question why we had additional arguments in these methods prior also.

Reply:
I believe this is safe to use `actionId` from `txMeta` since only provided `actionId` comes from `createTransactionEventFragment`.

But to be safe, and keep the scope as it is, I will rollback this change, and won't take any risk. We could hopefully revisit on MetricsController refactor.

Rolled back in https://github.com/MetaMask/metamask-extension/pull/21330/commits/b0706c0acd09ddbc45a726ee392eb6d732bcb090